lib/Db/NewsMapperV2.php

```diff
@@ -64,16 +64,31 @@ public function insert(Entity $entity): Entity
     }
 
     /**
-     * Remove deleted items.
+     * Remove deleted entities.
+     *
+     * @param string|null $userID       The user to purge
+     * @param int|null    $oldestDelete The timestamp to purge from
      *
      * @return void
      */
-    public function purgeDeleted(): void
+    public function purgeDeleted(?string $userID, ?int $oldestDelete): void
     {
         $builder = $this->db->getQueryBuilder();
-        $builder->delete($this->tableName)
-            ->where('deleted_at != 0')
-            ->execute();
+        $builder->delete($this->tableName);
+
+        if ($userID !== null) {
+            $builder->where('user_id = :user_id')
+                ->setParameter(':user_id', $userID);
+        }
+
+        if ($oldestDelete !== null) {
+            $builder->where('deleted_at < :deleted_at')
+                ->setParameter(':deleted_at', $oldestDelete);
+        } else {
+            $builder->where('deleted_at != 0');
+        }
```

Comment:
The last `->where` would override all previously set conditions ([QueryBuilder docs](https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/query-builder.html#high-level-api-methods)).
Apparently `->andWhere` could be used instead for every of them.

Reply:
😠 such a weird system

Reply:
Oh, wouldn't `deleted_at < :deleted_at` also delete items with `deleted_at = 0`?

Reply:
oh damn, good eye. I changed that code around a couple of times but missed that case I guess.